test_util/marathon.py

```diff
@@ -205,6 +205,83 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_pod(self, pod_definition, timeout=120):
+        """Deploy a pod to marathon
+
+        This function deploys an a pod and then waits for marathon to
+        aknowledge it's successfull creation or fails the test.
+
+        It waits until all the instances reach tasksRunning and then tasksHealthy state.
+
+        Args:
+            pod_definition: a dict with pod definition as specified in
+                            Marathon API
+            timeout: a time to wait for the pod to reach 'Healthy' status
+                     after which the test should be failed.
+            check_health: wait until Marathon reports tasks as healthy before
+                          returning
+        Returns:
+            Scaling instance count
+        """
+        r = self.post('v2/pods', json=pod_definition)
+        logging.info('Response from marathon: {}'.format(repr(r.json())))
+        assert r.ok
```

Comment:
must: This produces unusable / impossible to reason about what went wrong in the test result errors (all you get is r.ok was false, not what the status code was or the body which usually has useful info). Need to have an assertion comment which includes at least `r.status_code` and `r.body`. `r.raise_for_status()` is also an acceptable fix, although I prefer the more detailed assert